### ISSUE TITLE
Bug in GraphiQL when making a query of a row from a remote schema with Strapi

### ISSUE THREAD

**Issue by joshuarobs** [hasura/graphql-engine#4951]
I'm not sure if this is a bug with GraphiQL or Hasura or Strapi.

I managed to setup Hasura to use a Remote Schema from a local Strapi server which has its own GraphQL endpoint.

When I go to the Hasura console, in the GraphiQL tab, I get a bug if I try to click on the play button for a query that contains tables from the remote schema.

This is what I have in the box, my 2 queries:

````
query items {
  items {
    name
    brand {
      name
    }
  }
}
query users {
  test_users {
    id
    name
    email
    gender
  }
}
````

The `items` column is from Strapi, whereas `test_users` is a column from the Hasura postgres db, which can be found in the `Data` tab.

If I click on the play button and select the `users` query, it works perfectly. However if I click on the play button and select the `items` query, it gives me this error:

````
{
  "errors": [
    {
      "message": "Cannot query field \"test_users\" on type \"Query\". Did you mean \"users\"?",
      "locations": [
        {
          "line": 11,
          "column": 3
        }
      ],
      "extensions": {
        "code": "GRAPHQL_VALIDATION_FAILED",
        "exception": {
          "stacktrace": [
            "GraphQLError: Cannot query field \"test_users\" on type \"Query\". Did you mean \"users\"?",
            "    at Object.Field (/Users/x/WebstormProjects/strapi-cms/node_modules/graphql/validation/rules/FieldsOnCorrectType.js:53:31)",
            "    at Object.enter (/Users/x/WebstormProjects/strapi-cms/node_modules/graphql/language/visitor.js:324:29)",
            "    at Object.enter (/Users/x/WebstormProjects/strapi-cms/node_modules/graphql/language/visitor.js:375:25)",
            "    at visit (/Users/x/WebstormProjects/strapi-cms/node_modules/graphql/language/visitor.js:242:26)",
            "    at Object.validate (/Users/x/WebstormProjects/strapi-cms/node_modules/graphql/validation/validate.js:73:24)",
            "    at validate (/Users/x/WebstormProjects/strapi-cms/node_modules/apollo-server-core/dist/requestPipeline.js:221:34)",
            "    at Object.<anonymous> (/Users/x/WebstormProjects/strapi-cms/node_modules/apollo-server-core/dist/requestPipeline.js:118:42)",
            "    at Generator.next (<anonymous>)",
            "    at fulfilled (/Users/x/WebstormProjects/strapi-cms/node_modules/apollo-server-core/dist/requestPipeline.js:5:58)",
            "    at processTicksAndRejections (internal/process/task_queues.js:97:5)"
          ]
        }
      }
    }
  ]
}
```` 

So the query doesn't execute properly and doesn't give me the results from the server I'm looking for. The only workaround I had found was to delete all other queries except the one I'm running (if it involves tables from Strapi) and that would give me the data properly.

I'm assuming the cause of the bug is that somehow the other query is being run on the Strapi server, when it shouldn't, and that it fails because `test_users` doesn't exist on the remote schema (because it doesn't exist there and only exists in the Hasura db).

It looks like this may be a bug with Strapi, however, maybe other people may encounter a similar situation with other remote schemas, depending on their implementation.

Tl;dr: Hasura + Strapi graphQL as a remote schema. Running queries in GraphiQL with multiple queries in the box works as normal when attempting to run one of them, except for the query that involves tables/columns from the remote schema from Strapi. Only workaround is to delete all other queries temporarily, run the command and paste them back in.

**Comment by soorajshankar** [comment]
Hello @joshuarobs, Thanks for submitting the issue!

I was trying to reproduce this but unfortunately, I couldn't reproduce this.

> However if I click on the play button and select the `items` query, it gives me this error:

usually, when we do this, the console (front end app) will send a request that only contains the `items` query to the GQL engine. and I'm curious how the server returns with an error that refers to the `test_users`

I will need to check if the query contains anything related to the`test_users` when you select & play the `items` query.  Could you please share the network headers from the browser network tab while getting this error?

**Comment by beerose** [comment]
Closing because of inactivity.
